src/faldbt/lib.py

```diff
@@ -45,60 +64,122 @@ def _get_operation_node(manifest: Manifest, project_path, sql):
+        response, execute_result = adapter.execute(sql, fetch=True)
+
+        table = ResultTable(
+            column_names=list(execute_result.column_names),
+            rows=[list(row) for row in execute_result],
+        )
+
+        result = RemoteRunResult(
+            raw_sql=sql,
+            compiled_sql=sql,
+            node=node,
+            table=table,
+            timing=[],
+            logs=[],
+            generated_at=datetime.utcnow(),
+        )
+        adapter.connections.commit()
+
+    return response, result
+
+
+def _get_target_relation(
+    target: Union[ParsedModelNode, ParsedSourceDefinition], project_path: str
+):
+    adapter = _get_adapter(project_path)
 
-    return result
+    relation = None
+    with adapter.connection_named(str(uuid4())):
+        # This ROLLBACKs so it has to be a new connection
+        relation = adapter.get_relation(
+            target.database, target.schema, target.identifier
+        )
+    return relation
 
 
 def execute_sql(manifest: Manifest, project_path: str, sql: str):
-    adapter = _get_adapter(project_path)
-    return _execute_sql(manifest, project_path, adapter, sql)
+    _, result = _execute_sql(manifest, project_path, sql)
+    return result
 
 
-def fetch_model(manifest: Manifest, project_path: str, model: ParsedModelNode):
-    adapter = _get_adapter(project_path)
-    relation = adapter.get_relation(model.database, model.schema, model.identifier)
-    query = f'SELECT * FROM {relation}'
-    return _execute_sql(manifest, project_path, adapter, query)
+def fetch_target(
+    manifest: Manifest,
+    project_path: str,
+    target: Union[ParsedModelNode, ParsedSourceDefinition],
+):
+    relation = _get_target_relation(target, project_path)
+
+    if relation is None:
+        raise Exception(f"Could not get relation for '{target.unique_id}'")
+
+    query = f"SELECT * FROM {relation}"
+    _, result = _execute_sql(manifest, project_path, query)
+    return result
+
+
+def write_target(
+    data: pd.DataFrame,
+    manifest: Manifest,
+    project_path: str,
+    target: Union[ParsedModelNode, ParsedSourceDefinition],
+):
+    relation = _get_target_relation(target, project_path)
+
+    column_names: List[str] = list(data.columns)
+    rows = data.to_records(index=False)
+    row_dicts = list(map(lambda row: dict(zip(column_names, row)), rows))
+
+    agate_table = agate_helper.table_from_data(row_dicts, column_names=column_names)
+
+    # We are using the SQLAlchemy table to generate the SQL, but it targeting sqlite
+    # This may bite us later for other adapters
+    alchemy_table = agatesql.table.make_sql_table(
+        agate_table, table_name=target.identifier, db_schema=target.schema
+    )
+
+    if relation is None:
+        stmt = CreateTable(alchemy_table).compile(
+            compile_kwargs={"literal_binds": True}
+        )
+        stmt_str = six.text_type(stmt).strip()
+
+        _execute_sql(manifest, project_path, stmt_str)
```

Comment:
looks like these two lines are repeated below and can be after the if statement

Reply:
It is a different call, `stmt_str` is overwritten. Maybe I will change it.